src/job_cache/job_cache.cpp

```diff
@@ -326,7 +319,11 @@ FindJobResponse Cache::read(const FindJobRequest &find_request) {
       return *response;
     }
 
-    if (miss_on_failure && misses_from_failure > 30) {
+    if (response.error() == FindJobError::CouldNotConnect) {
```

Comment:
super  small nit that  I'm waffling on

`failed_on_connect |= response.error() == FindJobError::CouldNotConnect`

Reply:
I don't think that's worth doing personally but if you want it I can do it.